### DISCUSSION TITLE
Looking to query US State historical temp averages

### DISCUSSION THREAD

**Discussion by ericshively** [meteostat/meteostat-python#92] [General]
From initial API documentation, I was going to use Stations to fetch all stations in US, filter them by State column for later aggregating/averaging, then use Normals fetch for every single Station. I think there's like 1,000 stations, so is there a better way to batch this, or get averages across a US state? It would mostly be a 1-off query because I'd cache the data so that I wouldn't be hitting your API again on every request.

Essentially I'm looking to graph data that would be "For each US State, graph the avg. min, max, and avg. temp", similar to you how you do on https://meteostat.net but instead of 1 station, it's the average across all stations within a state.

**Comment by clampr** [comment]
Good question. At the moment you'd have to go through all the individual station dumps. That's possible (as you described above), but it takes some time for such a script to run.

However, the 100 km2 grids could be an option. They are not publicly documented, yet. But you can download them using this URL:

```
https://bulk.meteostat.net/v2/gridded/daily/{tavg|tmin|tmax|prcp|wspd|pres}/{YYYY-MM-DD}.nc
```

Example:

```
https://bulk.meteostat.net/v2/gridded/daily/tavg/2022-07-04.nc
```

Visualization could look like that:

<img width="820" alt="image" src="https://user-images.githubusercontent.com/12759785/177332961-e8f702c6-bc98-47dd-8196-2db59ec85644.png">
